lib/cli-engine/cli-engine.js

```diff
@@ -730,14 +730,8 @@ class CLIEngine {
         lastConfigArrays.length = 0;
 
         // Delete cache file; should this do here?
-        if (!cache) {
-            try {
-                fs.unlinkSync(cacheFilePath);
-            } catch (error) {
-                if (!error || error.code !== "ENOENT") {
-                    throw error;
-                }
-            }
+        if (!cache && fs.existsSync(cacheFilePath)) {
```

Comment:
I'm not sure this is sufficient enough to eliminate try/cache. File might exist, for example, but it might be locked. Or the process might not have permissions to delete a file. So I think try/cache is still needed.

Reply:
I like to make it ignoring more kinds of errors rather than checking something before the removing because file access is essentially parallel (I.e. other processes can manipulate the file while ESLint is doing).